tfx/orchestration/kubeflow/utils.py

```diff
@@ -20,17 +20,16 @@
 
 from tfx.dsl.components.base import base_node
 from tfx.orchestration import data_types
+from tfx.orchestration.kubeflow.v2.decorators import FinalStatusStr
 
 
 def replace_placeholder(component: base_node.BaseNode) -> None:
-  """Replaces the RuntimeParameter placeholders with kfp.dsl.PipelineParam."""
+  """Replaces the TFX placeholders with KFP placeholders"""
   keys = list(component.exec_properties.keys())
   for key in keys:
     exec_property = component.exec_properties[key]
-    if not isinstance(exec_property, data_types.RuntimeParameter):
-      continue
-    component.exec_properties[key] = str(
-        dsl.PipelineParam(name=exec_property.name))
+    if isinstance(exec_property, FinalStatusStr):
+      component.exec_properties[key] = '{{workflow.status}}'
```

Comment:
This function was only being used in base_component but it was not having any effect there, so moved it up and used it for FinalStatusStr

Reply:
This is for supporting runtime param if I remember correctly, could you create a separate function instead of changing this?

Reply:
I believe this function in the current codebase is a noop, you can see how below in the component those are not used. That said, I can derisk this by making it a separate function